effekt/jvm/src/main/scala/effekt/Runner.scala

```diff
@@ -87,13 +87,12 @@ trait Runner[Executable] {
   /**
    * Builds a given executable and returns the resulting path to the executable.
    */
-  def build(executable: Executable)(using Context): String
+  def build(executable: Executable)(using Context): Option[String]
 
   /**
    * Runs the executable (e.g. the main file) by calling the build function.
    */
-  def eval(executable: Executable)(using C: Context): Unit = {
-    val execFile = build(executable)
+  def eval(executable: Executable)(using C: Context): Unit = build(executable).map { execFile =>
```

Comment:
Nitpick: you can also use 
```suggestion
  def eval(executable: Executable)(using C: Context): Unit = build(executable).foreach { execFile =>
```
and then avoid the `.getOrElse(())`

(We use it elsewhere in the codebase too, I think?)